src/hooks/use-edit.ts

```diff
@@ -24,24 +25,27 @@ export const useEdit = ({
   moveCursorRight,
   bufferCommands,
   buffer,
+	setBuffer,
   setAppState,
   enabled
 }: EditParams) => {
   const [isMSN, setIsMSN] = useState(true);
 
   useEffect(() => {
     setIsMSN(true);
-  }, [cursor, buffer]);
```

Comment:
We can't reset the flag with each update, because we update the buffer when the user makes an edit.
Instead, `setIsMSN(true)` was called after each `bufferCommand` call at lines 63, 70, and 75.

Reply:
We can put the buffer back into this dependency array now